src/__tests__/fireEvent.test.tsx

```diff
@@ -475,3 +475,53 @@ test('has only onMove', () => {
   });
   expect(handleDrag).toHaveBeenCalled();
 });
+
+// Those events ideally should be triggered through `fireEvent.scroll`, but they are handled at the
+// native level, so we need to support manually triggering them
+describe('native events', () => {
+  test('triggers onScrollBeginDrag', () => {
+    const onScrollBeginDragSpy = jest.fn();
+    const { getByTestId } = render(
+      <ScrollView testID="test-id" onScrollBeginDrag={onScrollBeginDragSpy} />
+    );
+
+    fireEvent(getByTestId('test-id'), 'onScrollBeginDrag');
+    expect(onScrollBeginDragSpy).toHaveBeenCalled();
+  });
+
+  test('triggers onScrollEndDrag', () => {
+    const onScrollEndDragSpy = jest.fn();
+    const { getByTestId } = render(
+      <ScrollView testID="test-id" onScrollEndDrag={onScrollEndDragSpy} />
+    );
+
+    fireEvent(getByTestId('test-id'), 'onScrollEndDrag');
+    expect(onScrollEndDragSpy).toHaveBeenCalled();
+  });
+
+  test('triggers onMomentumScrollBegin', () => {
+    const onMomentumScrollBeginSpy = jest.fn();
+    const { getByTestId } = render(
+      <ScrollView
+        testID="test-id"
+        onMomentumScrollBegin={onMomentumScrollBeginSpy}
+      />
+    );
+
+    fireEvent(getByTestId('test-id'), 'onMomentumScrollBegin');
+    expect(onMomentumScrollBeginSpy).toHaveBeenCalled();
+  });
+
+  test('triggers onMomentumScrollEnd', () => {
+    const onMomentumScrollEndSpy = jest.fn();
+    const { getByTestId } = render(
+      <ScrollView
+        testID="test-id"
+        onMomentumScrollEnd={onMomentumScrollEndSpy}
+      />
+    );
+
+    fireEvent(getByTestId('test-id'), 'onMomentumScrollEnd');
+    expect(onMomentumScrollEndSpy).toHaveBeenCalled();
+  });
+});
```

Comment:
Should we need to add some other native events to this list?

Reply:
We could? I'm unsure why ones are relevant, I know that those ones are events that I use. The goal for this PR was to add some coverage to the feature, so didn't go further than that.

If we want to be exhaustive, we could also add something like `fireEvent(element, 'eventNameThatDoesNotExist)`, because such event would be triggered.